### PR TITLE
[CUBRIDQA-877] add cci compat test guide

### PR DESCRIPTION
**Pull request by slamdunkorchid**
please refer to http://jira.cubrid.org/browse/CUBRIDQA-877
After cci guide finished, I'll update it again

### CODE REVIEW

File: doc/cci_compatibility_guide.md [hunk 1]
```diff
@@ -0,0 +1,730 @@
...
+    default.cubrid.cubrid_port_id=8220
+    default.broker1.BROKER_PORT=8420
+    default.broker1.APPL_SERVER_SHM_ID=8420
+    default.broker2.BROKER_PORT=8520
+    default.broker2.APPL_SERVER_SHM_ID=8520
+    ...
+   ```
+
+* Install CQT
+  * checkout CQT
```
Comment (fanzaiqiang) [inline doc/cci_compatibility_guide.md]:
```suggestion
  * Check out CQT
```

File: doc/cci_compatibility_guide.md [hunk 2]
```diff
@@ -0,0 +1,730 @@
...
+ ![fail reason](./cci_compatibility_image/fail_details.PNG)
+ 
+ Above told us that, when program can not connect to server ,sometime it will appears -191 error, it is not a bug. So we need modify test cases.    
+
+ Refer to [Check failure detail of cci guide](https://github.com/slamdunkorchid/cubrid-testtools/blob/0815/doc/cci_guide.md#check-failure-detail)
+    
+# 5. CCI Compatibility Test Case 
+## Test Case Version
+CCI compatibility test cases are the same with CCI test cases. But we need to decide the version of test case. it based on current server. For example:
+* low version's server
```
Comment (fanzaiqiang) [inline doc/cci_compatibility_guide.md]:
```suggestion
* Compat with low version server  
```

File: doc/cci_compatibility_guide.md [hunk 3]
```diff
@@ -0,0 +1,730 @@
...
+If we test 10.2's driver with 8.4.3's server, we need use [8.4.3's test cases](https://oss.navercorp.com/CUBRID/cubridqa/tree/RB-8.4.3/interface/CCI/shell/_20_cci)   
+If we test 10.2's driver with 8.4.4's server, we need use [8.4.4's test cases](https://oss.navercorp.com/CUBRID/cubridqa/tree/RB-8.4.4/interface/CCI/shell/_20_cci)   
+If we test 10.2's driver with 9.2's server, we need use [9.2's test cases](https://oss.navercorp.com/CUBRID/cubridqa/tree/RB-9.2.0/interface/CCI/shell/_20_cci)   
+If we test 10.2's driver with 9.3's server, we need use [9.3's test cases](https://oss.navercorp.com/CUBRID/cubridqa/tree/RB-9.3.0/interface/CCI/shell/_20_cci)    
+If we test 10.2's driver with 10.0's server, we need use [10.0's test cases](https://github.com/CUBRID/cubrid-testcases-private/tree/release/10.0/interface/CCI/shell)   
+If we test 10.2's driver with 10.1's server, we need use [10.1's test cases](https://github.com/CUBRID/cubrid-testcases-private/tree/release/10.1/interface/CCI/shell)     
+
+* low version's driver
```
Comment (fanzaiqiang) [inline doc/cci_compatibility_guide.md]:
```suggestion
* Compat with low version driver  
```

File: doc/cci_compatibility_guide.md [hunk 4]
```diff
@@ -0,0 +1,730 @@
...
+If we test 10.2's server with 8.4.1/8.4.3/8.4.4/9.2/9.3/10.0/10.1's driver, we need use [10.2's test cases](https://github.com/CUBRID/cubrid-testcases-private/tree/develop/interface/CCI/shell)     
+
+## Test Case Specification
+test cases specification, please refer to [cci test cases](https://github.com/slamdunkorchid/cubrid-testtools/blob/0815/doc/cci_guide.md#6-cci-test-case)
```
Comment (fanzaiqiang) [inline doc/cci_compatibility_guide.md]:
```suggestion
Regarding as test case specification, please refer to [CCI test cases](https://github.com/slamdunkorchid/cubrid-testtools/blob/0815/doc/cci_guide.md#6-cci-test-case)
```

File: doc/cci_compatibility_guide.md [hunk 5]
```diff
@@ -0,0 +1,730 @@
...
+```
+
+## 2.4 Execute test      
+Please refer to [2.4 Execute test of cci guide](https://github.com/slamdunkorchid/cubrid-testtools/blob/0815/doc/cci_guide.md#24-execute-test)
+
+## 2.5 Examine test results
+Please refer to [2.5 Examine test results of cci guide](https://github.com/slamdunkorchid/cubrid-testtools/blob/0815/doc/cci_guide.md#25-examine-test-results)
+
+# 3. Deploy Regression Test Environment
+## 3.1 Test Machine
```
[the remark this answers is not in the document]
Comment (slamdunkorchid) [reply doc/cci_compatibility_guide.md]: all the other guide are writted with test Machine, and this test have server and driver  test, avoid confusion
I think no need to change
Test Machine --> Test Machines 
like https://github.com/CUBRID/cubrid-testtools/blob/develop/doc/isolation_guide.md#31-test-machines

**Comment by slamdunkorchid** [comment]
please continue to view

File: doc/cci_compatibility_guide.md [hunk 6]
```diff
@@ -0,0 +1,730 @@
...
+When a new build comes, the test will start. We just need to make sure that test environment has no problem and listener has been started. Sometimes, in order to investigate or correct a test, we need to send messages manually.
+### Start the listener
+```bash
+$ cd ~
+$ sh start_test.sh &
+$ tail -f nohup.out
+```
+### Send test message
+#### Configuration 
+There are configuration files for compatibility messages
```
Comment (fanzaiqiang) [inline doc/cci_compatibility_guide.md]: We need add  full configuration about CCI (also JDBC) compatiblity test messages.

* CTP/conf/job.conf

		job_compat.service=ON
		job_compat.crontab=0/5 * * * * ?
		job_compat.listenfile=CUBRID-{1}-linux.x86_64.sh
		job_compat.acceptversions=10.0.*.0~8999,10.1.*,10.2.*
		job_compat.package_bits=64
		job_compat.package_type=general

		...
		...

		job_compat.test.3.scenario=compat_cci_D64
		job_compat.test.3.queue=QUEUE_CUBRID_QA_COMPAT_CCI_SHELL_DRIVER_64
		job_compat.test.3.ext_config=compat.conf
		job_compat.test.3.ext_keys=cci_compatibility_for_{version}_D64

		job_compat.test.4.scenario=compat_cci_S64
		job_compat.test.4.queue=QUEUE_CUBRID_QA_COMPAT_CCI_SHELL_SERVER_64
		job_compat.test.4.ext_config=compat.conf
		job_compat.test.4.ext_keys=cci_compatibility_for_{version}_S64

		...
		...


* CTP/conf/compat.conf

		EXT_KEY_FOR_FIX_MAX=COMPAT_BUILD_ID
		EXT_KEY_FOR_FIX_MAX_FOLLOW=COMPAT_BUILD_URLS
		EXT_KEY_FOR_FIX_MAX_FOLLOW_KR=COMPAT_BUILD_URLS_KR
		EXT_KEY_FOR_TEST_CATAGORY=COMPAT_TEST_CATAGORY

		...
		...

		cci_compatibility_for_10.2.0_D64=cci_shell_10.1_S64.msg,cci_shell_10.0_S64.msg

		cci_compatibility_for_10.1.0_D64=cci_shell_10.2_S64.msg,cci_shell_10.0_S64.msg

		cci_compatibility_for_10.0.0_D64=cci_shell_10.2_S64.msg,cci_shell_10.1_S64.msg

		cci_compatibility_for_10.2.0_S64=cci_shell_10.1_D64.msg,cci_shell_10.0_D64.msg,cci_shell_9.3_D64.msg,cci_shell_9.2_D64.msg,cci_shell_8.4.4_D64.msg,cci_shell_8.4.3_D64.msg,cci_shell_8.4.1_D64.msg,

		cci_compatibility_for_10.1.0_S64=cci_shell_10.2_D64.msg,cci_shell_10.0_D64.msg,cci_shell_9.3_D64.msg,cci_shell_9.2_D64.msg,cci_shell_8.4.4_D64.msg,cci_shell_8.4.3_D64.msg,cci_shell_8.4.1_D64.msg,

		cci_compatibility_for_10.0.0_S64=cci_shell_10.2_D64.msg,cci_shell_10.1_D64.msg,cci_shell_9.3_D64.msg,cci_shell_9.2_D64.msg,cci_shell_8.4.4_D64.msg,cci_shell_8.4.3_D64.msg,cci_shell_8.4.1_D64.msg,


* Detail compat configuration

		~/CTP/conf/compat/cci_shell_8.4.3_D64.msg
		COMPAT_BUILD_ID=8.4.3.{max}
		COMPAT_BUILD_URLS=CUBRID-{BUILD_ID}-linux.x86_64.sh
		COMPAT_BUILD_BIT=64
		COMPAT_BUILD_TYPE=general
		COMPAT_BUILD_SCENARIOS=cci

		~/CTP/conf/compat/cci_shell_10.0_S64.msg
		COMPAT_BUILD_ID=10.0.{dmax}
		COMPAT_BUILD_URLS=CUBRID-{BUILD_ID}-linux.x86_64.sh
		COMPAT_BUILD_BIT=64
		COMPAT_BUILD_TYPE=general
		COMPAT_BUILD_SVN_BRANCH=unknown
		COMPAT_BUILD_SCENARIOS=cci
		COMPAT_BUILD_SCENARIO_BRANCH_GIT=release/10.0

		~/CTP/conf/compat/cci_shell_10.1_D64.msg
		COMPAT_BUILD_ID=10.1.{dmax}
		COMPAT_BUILD_URLS=CUBRID-{BUILD_ID}-Linux.x86_64.sh
		COMPAT_BUILD_BIT=64
		COMPAT_BUILD_TYPE=general
		COMPAT_BUILD_SCENARIOS=cci

		~/CTP/conf/compat/cci_shell_10.2_D64.msg
		COMPAT_BUILD_ID=10.2.{dmax}
		COMPAT_BUILD_URLS=CUBRID-{BUILD_ID}-Linux.x86_64.sh
		COMPAT_BUILD_BIT=64
		COMPAT_BUILD_TYPE=general
		COMPAT_BUILD_SCENARIOS=cci

		~/CTP/conf/compat/cci_shell_9.1_D64.msg
		COMPAT_BUILD_ID=9.1.0.0212
		COMPAT_BUILD_URLS=CUBRID-{BUILD_ID}-linux.x86_64.sh
		COMPAT_BUILD_BIT=64
		COMPAT_BUILD_TYPE=general
		COMPAT_BUILD_SCENARIOS=cci

		~/CTP/conf/compat/cci_shell_9.2_D64.msg
		COMPAT_BUILD_ID=9.2.{dmax}
		COMPAT_BUILD_URLS=CUBRID-{BUILD_ID}-linux.x86_64.sh
		COMPAT_BUILD_BIT=64
		COMPAT_BUILD_TYPE=general
		COMPAT_BUILD_SCENARIOS=cci

		~/CTP/conf/compat/cci_shell_10.1_S64.msg
		COMPAT_BUILD_ID=10.1.{dmax}
		COMPAT_BUILD_URLS=CUBRID-{BUILD_ID}-Linux.x86_64.sh
		COMPAT_BUILD_BIT=64
		COMPAT_BUILD_TYPE=general
		COMPAT_BUILD_SVN_BRANCH=unknown
		COMPAT_BUILD_SCENARIOS=cci
		COMPAT_BUILD_SCENARIO_BRANCH_GIT=release/10.1

		~/CTP/conf/compat/cci_shell_10.2_S64.msg
		COMPAT_BUILD_ID=10.2.{dmax}
		COMPAT_BUILD_URLS=CUBRID-{BUILD_ID}-Linux.x86_64.sh
		COMPAT_BUILD_BIT=64
		COMPAT_BUILD_TYPE=general
		COMPAT_BUILD_SVN_BRANCH=unknown
		COMPAT_BUILD_SCENARIOS=cci
		COMPAT_BUILD_SCENARIO_BRANCH_GIT=develop

		~/CTP/conf/compat/cci_shell_10.0_D64.msg
		COMPAT_BUILD_ID=10.0.{dmax}
		COMPAT_BUILD_URLS=CUBRID-{BUILD_ID}-linux.x86_64.sh
		COMPAT_BUILD_BIT=64
		COMPAT_BUILD_TYPE=general
		COMPAT_BUILD_SCENARIOS=cci

		~/CTP/conf/compat/cci_shell_9.0_D64.msg
		COMPAT_BUILD_ID=9.0.{dmax}
		COMPAT_BUILD_URLS=CUBRID-{BUILD_ID}-linux.x86_64.sh
		COMPAT_BUILD_BIT=64
		COMPAT_BUILD_TYPE=general
		COMPAT_BUILD_SCENARIOS=cci

		~/CTP/conf/compat/cci_shell_8.4.5_D64.msg
		COMPAT_BUILD_ID=8.4.5.{max}
		COMPAT_BUILD_URLS=CUBRID-{BUILD_ID}-linux.x86_64.sh
		COMPAT_BUILD_BIT=64
		COMPAT_BUILD_TYPE=general
		COMPAT_BUILD_SCENARIOS=cci

		~/CTP/conf/compat/cci_shell_8.4.4_D64.msg
		COMPAT_BUILD_ID=8.4.4.{max}
		COMPAT_BUILD_URLS=CUBRID-{BUILD_ID}-linux.x86_64.sh
		COMPAT_BUILD_BIT=64
		COMPAT_BUILD_TYPE=general
		COMPAT_BUILD_SCENARIOS=cci

		~/CTP/conf/compat/cci_shell_9.3_D64.msg
		COMPAT_BUILD_ID=9.3.{dmax}
		COMPAT_BUILD_URLS=CUBRID-{BUILD_ID}-linux.x86_64.sh
		COMPAT_BUILD_BIT=64
		COMPAT_BUILD_TYPE=general
		COMPAT_BUILD_SCENARIOS=cci

		~/CTP/conf/compat/cci_shell_9.4_D64.msg
		COMPAT_BUILD_ID=9.4.{dmax}
		COMPAT_BUILD_URLS=CUBRID-{BUILD_ID}-linux.x86_64.sh
		COMPAT_BUILD_BIT=64
		COMPAT_BUILD_TYPE=general
		COMPAT_BUILD_SCENARIOS=cci

		~/CTP/conf/compat/cci_shell_8.4.1_D64.msg
		COMPAT_BUILD_ID=8.4.1.{max}
		COMPAT_BUILD_URLS=CUBRID-{BUILD_ID}-linux.x86_64.sh
		COMPAT_BUILD_BIT=64
		COMPAT_BUILD_TYPE=general
		COMPAT_BUILD_SCENARIOS=cci

File: doc/cci_compatibility_guide.md [hunk 7]
```diff
@@ -0,0 +1,730 @@
...
+cci_shell_9.3_S64.msg
+cci_shell_9.4_D64.msg
+cci_shell_9.4_S64.msg
+```
+cci_shell_10.0_D64.msg - it means test current build's server with 10.0's driver   
+cci_shell_10.0_S64.msg - it means test current build's driver with 10.0's server   
+cci_shell_8.4.1_S64.msg - it means test current build's driver with 8.4.1's server   
+cci_shell_8.4.1_D64.msg - it means test current build's server with 8.4.1's driver    
+ 
+#### Current server test
```
Comment (fanzaiqiang) [inline doc/cci_compatibility_guide.md]:
```suggestion
#### Send test messages for server test
```

File: doc/cci_compatibility_guide.md [hunk 8]
```diff
@@ -0,0 +1,730 @@
...
+We perform cci compatibility test for each build.     
+## 4.1 Daily Regresion Test
+When a new build comes, the test will start. We just need to make sure that test environment has no problem and listener has been started. Sometimes, in order to investigate or correct a test, we need to send messages manually.
+### Start the listener
+```bash
+$ cd ~
+$ sh start_test.sh &
+$ tail -f nohup.out
+```
+### Send test message
```
Comment (fanzaiqiang) [inline doc/cci_compatibility_guide.md]: Please use indent format.
It's better to have intuitive hierarchy.

File: doc/cci_compatibility_guide.md [hunk 9]
```diff
@@ -0,0 +1,730 @@
...
+   ```
+    default.cubrid.cubrid_port_id=8220
+    default.broker1.BROKER_PORT=8420
+    default.broker1.APPL_SERVER_SHM_ID=8420
+    default.broker2.BROKER_PORT=8520
+    default.broker2.APPL_SERVER_SHM_ID=8520
+    ...
+   ```
+
+* Install CQT
```
[the remark this answers is not in the document]
Comment (slamdunkorchid) [reply doc/cci_compatibility_guide.md]: If delete them, we don't test 10.2 driver and 8.4.x server and 9.x  server
Now the public test environment don't  test them .
If you need CQT and oss no longer, you can delete them.

[the remark this answers is not in the document]
Comment (slamdunkorchid) [reply doc/cci_compatibility_guide.md]: current public jdbc compat encironment  don't deploy CQT and oss test cases,
it is still running
>=10.0 driver
>=10.0 server

[the remark this answers is not in the document]
Comment (fanzaiqiang) [reply doc/cci_compatibility_guide.md]: So let's delete description about CQT and checking out test cases from oss.

**Comment by slamdunkorchid** [comment]
please continue to view

**Comment by fanzaiqiang** [comment]
Please submit an image file I provide to you. Then, this will pass final review. And close it.

**Pull request merged**